File: seed/php-sdk/trace/src/V2/Problem/Types/Parameter.php
<?php

namespace Seed\V2\Problem\Types;

use Seed\Core\SerializableType;
use Seed\Core\JsonProperty;

class Parameter extends SerializableType
{
    /**
     * @var string $parameterId
     */
    #[JsonProperty("parameterId")]
    public string $parameterId;

    /**
     * @var string $name
     */
    #[JsonProperty("name")]
    public string $name;

    /**
     * @var mixed $variableType
     */
    #[JsonProperty("variableType")]
    public mixed $variableType;

    /**
     * @param array{
     *   parameterId: string,
     *   name: string,
     *   variableType: mixed,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->parameterId = $values['parameterId'];
        $this->name = $values['name'];
        $this->variableType = $values['variableType'];
    }
}

File: seed/ruby-sdk-v2/trace/lib/seed/v_2/v_3/problem/types/parameter.rb
# frozen_string_literal: true

module Seed
  module V2
    module V3
      module Problem
        module Types
          class Parameter < Internal::Types::Model
            field :parameter_id, -> { String }, optional: false, nullable: false
            field :name, -> { String }, optional: false, nullable: false
            field :variable_type, -> { Seed::Commons::Types::VariableType }, optional: false, nullable: false
          end
        end
      end
    end
  end
end
